### DISCUSSION TITLE
Options for an invite-only service

### DISCUSSION THREAD

**Discussion by caschoener** [fief-dev/fief#359] [General]
I'd like to make my service invite only, but I'm pretty open-minded as to how it's done. Additionally I want users to be federated into different organizations.  I know there isn't an email invite system.  

I was considering just giving people a secret token that would register them, and ideally automatically map them to their organization.  It seems like codebase doesn't support server-side validation of signup fields like this though.  Is that accurate?  I could consider forking and building the feature myself.

I love the overall codebase and featureset so I'd love to make it work.

**Comment by caschoener** [comment]
I also wonder, is this a use case for tenants?  I've read through the docs for multi tenancy but it's not clear to me how to actually auto-place users into different tenants with some validation

**Comment by frankie567** [comment]
Right now, we indeed don't have an invite process; but it's on our long-term roadmap.

A workaround to do what you want would be to require a [permission](https://docs.fief.dev/configure/access-control/) everywhere in your application. By default, users won't have this permission. However, when processing the Fief's callback, you can check if it matches your rules (email domain name, secret token...) and grant them the permission.

---

Tenants are a way to separate users in different sub-apps. It's a good use-case if you sell your app in white-label. The important nuance is that a same person can have one account per tenant, with the same email address. From their perspective, it's like a different app/service.

So I'm not sure it's what you want to do when you talk about "organizations".
